### DISCUSSION TITLE
How to get freshrss client id

### DISCUSSION THREAD

**Discussion by neilstottler** [FreshRSS/FreshRSS#6974] [Q&A]
Hi, my freshrss feed is getting rate limited by xcancel.com and they are asking for my client request ID. How do I get this? Thanks!

**Comment by Alkarex** [comment]
There is no such thing as a FreshRSS client ID

**Reply by neilstottler** [reply]
Ok, can you point me where to see the rss file that is downloaded through freshrss when connecting a feed? That ID is stored in there in this case.

Here's an example of what I mean in an rss file that can be downloaded using wget.
```<?xml version="1.0" encoding="UTF-8"?>
  <rss xmlns:atom="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/elements/1.1/" version="2.0">
    <channel>
      <atom:link href="https://rss.xcancel.com/nasa/rss" rel="self" type="application/rss+xml" />
      <title>RSS reader not yet whitelisted!</title>
      <link>https://rss.xcancel.com/nasa/rss</link>
      <description>RSS reader not yet whitelist! Please send an email rss [AT] xcancel [DOT] com with this ID to get your RSS feed reader whitelisted: IDHERE</description>
      <language>en-us</language>
      <ttl>1</ttl>
        <item>
          <title>RSS reader not yet whitelisted!</title>
          <dc:creator>@xcancel</dc:creator>
          <description>RSS reader not yet whitelist! Plain request with just the ID will be ignored! Please send an email rss [AT] xcancel [DOT] com with the reason why you want your RSS feed reader to be whitelist and this ID: IDHERE</description>
          <pubDate>Mon, 01 January 1971 00:00:00 GMT</pubDate>
          <guid>https://rss.xcancel.com/nasa/rss</guid>
          <link>https://rss.xcancel.com/nasa/rss</link>
        </item>
    </channel>
  </rss>
```

**Comment by neilstottler** [comment]
Allow me to rephrase this. Is there a way I can verbosely use freshrss on a source? I have a source that is rate limiting me and I need to supply them with an ID that it returned in a rss file. They specifically are asking for me to do it through freshrss and not something like wget/curl.

**Reply by Alkarex** [reply]
Depending on how you installed FreshRSS, you can grep into the `FreshRSS/data/cache/` directory 

```sh
cd data/cache/
grep -B 10 -A 10 -i 'xcancel.com' -r .
```

This should allow you to find the cache version of the corresponding feed
